mongo/cursor_test.go

```diff
@@ -20,9 +20,10 @@ type testBatchCursor struct {
 	batches []*bsoncore.DocumentSequence
 	batch   *bsoncore.DocumentSequence
 	closed  bool
+	nextErr error
```

Comment:
IIUC, this isn't actually used now that we've switched to integration tests. If that's the case, the changes in this file should be reverted.